SwiftRuntimeLibrary/SwiftMarshal/SwiftProtocolTypeAttribute.cs

```diff
@@ -49,6 +49,15 @@ public static Type ProxyTypeForInterfaceType (Type interfaceType)
 			return proxyAttr.ProxyType;
 		}
 
+		public static T MakeProxy<T> (Type proxyType, T interfaceImpl)
+		{
+			var interfaceType = typeof (T);
+			var ci = proxyType.GetConstructor (new Type [] { interfaceType });
+			if (ci == null)
+				throw new SwiftRuntimeException ($"Type {proxyType.Name} does not have a constructor that takes takes {interfaceType.Name}");
+			return (T)ci.Invoke (new object [] { interfaceImpl });
```

Comment:
Is this cast always going to work?

Reply:
Yes - it's the interface type - the proxy implements that interface. It has to work. Great question!